File: Src/Domain/FutureSaving.cs
﻿namespace Domain;

/// <summary>
/// Savings that user has planned for future
/// </summary>

public class FutureSaving
{
    public Guid Id { get; set; }
    public decimal Amount { get; set; }

    public decimal CompletedAmount { get; set; }

    public DateTime Date { get; set; }

    public Goal Goal { get; set; }

    public Guid FromAccountId { get; set; }
    public Account FromAccount { get; set; }
    public Guid ToAccountId { get; set; }
    public Account ToAccount { get; set; }

    public IEnumerable<Saving> CompletedSavings { get; set; } = new List<Saving>();

    public Guid BudgetId { get; set; }
    public Budget Budget { get; set; }
}

File: Src/WebUI/src/app/models/spendingPlan/futureSaving.ts
import { Saving } from "../saving"


export interface FutureSaving {
  id: string
  amount: number
  completedAmount: number
  date: Date | null
  fromAccountName: string
  toAccountName: string
  goalName?: string | null
  completedSavings: Saving[]
}

export class FutureSaving implements FutureSaving {
  constructor(init?: FutureSavingFormValues) {
    Object.assign(this, init)
  }
}

export class FutureSavingFormValues {
  id?: string = undefined
  amount: number = 0
  date: Date | null = null
  fromAccountName: string = ''
  toAccountName: string = ''
  goalName: string | null = null
}